resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/rest/kubernetes/v2/ResourceStagingServerSslOptionsProviderSuite.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.rest.kubernetes.v2
+
+import java.io.{File, FileInputStream, StringWriter}
+import java.security.KeyStore
+
+import com.google.common.base.Charsets
+import com.google.common.io.Files
+import org.bouncycastle.openssl.jcajce.JcaPEMWriter
+import org.scalatest.BeforeAndAfter
+
+import org.apache.spark.{SparkConf, SparkFunSuite, SSLOptions}
+import org.apache.spark.deploy.kubernetes.SSLUtils
+import org.apache.spark.util.Utils
+
+class ResourceStagingServerSslOptionsProviderSuite extends SparkFunSuite with BeforeAndAfter {
+
+  private val SSL_TEMP_DIR = Utils.createTempDir(namePrefix = "resource-staging-server-ssl-test")
+  private val KEYSTORE_FILE = new File(SSL_TEMP_DIR, "keyStore.jks")
+
+  private var sparkConf: SparkConf = _
+  private var sslOptionsProvider: ResourceStagingServerSslOptionsProvider = _
+
+  before {
+    sparkConf = new SparkConf(true)
+    sslOptionsProvider = new ResourceStagingServerSslOptionsProviderImpl(sparkConf)
+  }
+
```

Comment:
clear the `SSL_TEMP_DIR` between tests to reduce risk of polluting other test runs